### ISSUE TITLE
Add xml comments for all public members

### ISSUE THREAD

**Issue by Shane32** [graphql-dotnet/graphql-dotnet#1857]
I think there should be a simple `<summary>` tag for each public member (all CS1591 build warnings) by the time 4.0 is released.  This issue will track progress as additional documentation is added to 3.x versions.

**Comment by sungam3r** [comment]
This is too general and a big task. It is just impossible to make a PR fixing that issue because of its size and time spent.

**Comment by Shane32** [comment]
This issue is for tracking within the milestone and progress, much like the issue regarding error handling, which we finished over several pull requests.  I'll probably add checkboxes in the description above for the different namespaces to track progress.  I'll post individual PRs for different classes or sets of classes as I have time.

If you also have large goals, I suggest you create similar issues to track progress and visibility from the milestone.

**Comment by Shane32** [comment]
I actually had it outlined already, but the browser refreshed, and I lost it; half hour's work gone.....

**Comment by sungam3r** [comment]
Maybe use [projects](https://github.com/graphql-dotnet/graphql-dotnet/projects) for this? I haven't tried it yet.

**Comment by Shane32** [comment]
We have use projects within our company. I don’t think it’s a fit here. It’s basically another sorting technique like milestones. So if you had multiple teams working in one repo each team can have their own board and monitor progress. It doesn’t work well enough to fully replace other job management software unfortunately. We ended up just using it to group issues by topic (which could be done with tags just as well).

**Comment by sungam3r** [comment]
I see 75 warnings for CS1591 after disabling it in StarWars sample project and benchmarks (#1979). Not so much but not so little, especially if you write thoughtful and useful comments.

**Comment by Shane32** [comment]
I really do want to work on this.

**Comment by Shane32** [comment]
I’m not so concerned about the comments being so very thought out and detailed, but just any sort of comment is 100% improvement. Of course I don’t want to do a shoddy job either.

**Comment by sungam3r** [comment]
I messed something up in my previous comment about warnings. Actually I see ~1600 CS1591 warnings 😞

**Comment by Shane32** [comment]
It seems that methods that inherit xml documentation from the interface method they implement still generate a CS1591 warning.  I'm not sure we want to add `<inheritdoc/>` tags everywhere when VS already infers the doc inheritance.  But unless we do, it will show many more warnings than really exist.  What do you think @sungam3r  ?

**Comment by sungam3r** [comment]
I would go down the straight path - add comments where they are needed first, then place `<inheritdoc/>` somewhere else. In addition, sometimes in a derived class a comment should contain more information, and not just `<inheritdoc/>`.

**Comment by Shane32** [comment]
> I messed something up in my previous comment about warnings. Actually I see ~1600 CS1591 warnings 😞

Down to around 1050.  I think nearly half of those are within the `Language` namespace.

**Comment by Shane32** [comment]
Tomorrow I'll probably work on the introspection fields and misc/utility classes.  Those will take the longest, as each one will be different.  Maybe I'll see if I can tackle the Language namespace afterwards -- hopefully I can buzz through it "quickly", but with 500 members that needs commenting, it's still a lot of work.  That leaves a few leftover pieces that I have not worked on:

- Subscriptions and related (event stream resolvers, subscription methods on field builders, etc)
- Schema builder
- Schema printer

If you'd like to assist with these parts of the library, please do @sungam3r .  I feel that at least the subscriptions portion should be completely done, if possible, before 4.0.  The other two we should at least provide xml comments for the entry points (constructors, stuff people would commonly use).  Of course I'd really like to get to 100% if we can.

**Comment by Shane32** [comment]
Down to 623 warnings. Finished the `Language` namespace but didn't do the misc classes yet.

**Comment by sungam3r** [comment]
@Shane32 Transfer to v5?

**Comment by Shane32** [comment]
No, it should be a goal for 4.x, but doesn't need to be any more complete for 4.0.

**Comment by Shane32** [comment]
@sungam3r We only have 163 missing xml comments left.  I have updated the list at the top of this issue.

**Comment by Shane32** [comment]
@sungam3r Only classes that relate to GraphQL federation are left

@joemcbride If you want to contribute, perhaps you would be interested in adding XML comments to the classes and methods that relate to GraphQL Federation, as listed above.  About 25 comments are needed to cover all the public members.  We have completed all other XML comments, excluding the SchemaPrinter, as it is planned to rebuild it around the new AST printer in the Parser library.

**Comment by sungam3r** [comment]
It seems to be that I'll add all comments if I can to fight with code-first federation. Too less free time the last days.
